homeassistant/components/axis/hub/entity_loader.py

```diff
@@ -42,37 +44,39 @@ def register_platform(
         descriptions: tuple[AxisEventDescription, ...],
     ) -> None:
         """Register Axis entity platforms."""
-        self.platforms.append((async_add_entities, entity_class, descriptions))
+        topics: tuple[EventTopic, ...]
+        for description in descriptions:
+            if isinstance(description.event_topic, EventTopic):
+                topics = (description.event_topic,)
+            else:
+                topics = description.event_topic
+            for topic in topics:
+                self.topic_to_entity.setdefault(topic, []).append(
+                    (async_add_entities, entity_class, description)
+                )
 
     @callback
-    def initialize_platforms(self) -> None:
-        """Prepare event listeners and platforms."""
-
-        @callback
-        def load_entities(
-            platform_entity: type[AxisEventEntity],
-            descriptions: tuple[AxisEventDescription, ...],
-            async_add_entities: AddEntitiesCallback,
-        ) -> None:
-            """Set up listeners for events."""
-
-            @callback
-            def create_entity(description: AxisEventDescription, event: Event) -> None:
-                """Create Axis entity."""
-                event_id = (event.topic, event.topic_base, event.id)
-                if event_id in self.registered_events:
-                    # Device has restarted and all events are initiatlized anew
-                    return
-                self.registered_events.add(event_id)
-                if description.supported_fn(self.hub, event):
-                    async_add_entities([platform_entity(self.hub, description, event)])
+    def _create_entities_from_event(self, event: Event) -> None:
+        """Create Axis entities from event."""
+        event_id = (event.topic, event.topic_base, event.id)
+        if event_id in self.registered_events:
+            # Device has restarted and all events are initiatlized anew
+            return
+        self.registered_events.add(event_id)
+        for (
+            async_add_entities,
```

Comment:
It looks like they are all going to be different callbacks, but if not it would be good to group them to minimize the number of tasks